actioncable/lib/action_cable/engine.rb

```diff
@@ -40,5 +42,24 @@ class Railtie < Rails::Engine # :nodoc:
         options.each { |k,v| send("#{k}=", v) }
       end
     end
+
+    initializer "action_cable.routes" do
+      config.after_initialize do |app|
+        config = app.config
+        unless config.action_cable.mount_path.nil?
+          app.routes.prepend do
+            mount ActionCable.server => config.action_cable.mount_path
+          end
+        end
+
+        ActionDispatch::Routing::RouteWrapper.class_eval do
+          class_attribute :cable_mount_path_regex
+          prepend ActionCable::RouteWrapper
+          self.cable_mount_path_regex = %r{\A#{config.action_cable.mount_path}\z}
+        end
```

Comment:
This is to mark Action Cable's route as "internal". I have a PR open to try and add an option to the router API to do this in a less monkey-patching way.